packages/ember-data/lib/system/model/model.js

```diff
@@ -777,7 +793,8 @@ var Model = Ember.Object.extend(Ember.Evented, {
     });
     //We use the pathway of setting the hasMany as if it came from the adapter
     //because the user told us that they know this relationships exists already
-    this._relationships[key].updateRecordsFromAdapter(recordsToSet);
+    this.get(key);
```

Comment:
The relationship may not be properly instantiated at this point, using get will create it if it does not exist it seems